kasa/httpclient.py

```diff
@@ -49,6 +49,11 @@ async def post(
         response_data = None
         self._last_url = url
         self.client.cookie_jar.clear()
+        return_json = bool(json)
+        # If json is not a dict send as data
```

Comment:
Adding a reason for doing such a hack (instead of fixing the passed value on call-site) would make this more understandable. I just wrote a comment about fixing this at the call-sites, but this is here as the json-param contains a generator, right?

Reply:
Yes.  So `aiohttp` only accepts the generator as `data` but the way our client works is if it gets json it returns json back.  So this allows the generator to be passed with the `json` parameter.

Reply:
Yeah, I meant that it'd be great to document the behavior here, as I was first reading this and asking myself why wouldn't we fix this at the call-sites :-)

Reply:
Done